### PR TITLE
Added more query cache test for binding tuple value expression

### PR DESCRIPTION
**Pull request by nwang57**
Talked with @chenboy regarding #1298 , he prefers manually call `SetBoundOid()` to bind the tuple value expression in order to make the expression equality check work.

* Added a test to query cache test

### CODE REVIEW

**Comment by nwang57** [comment]
Yeah, thanks @saatviks . I can't change the label tho.

**Comment by saatviks** [comment]
@chenboy waiting on your review to this PR?

File: test/codegen/query_cache_test.cpp [hunk 1]
```diff
@@ -51,6 +51,30 @@ class QueryCacheTest : public PelotonCodeGenTest {
         &GetTestTable(TestTableId()), a_gt_40, {0, 1}));
   }
 
+  std::shared_ptr<planner::SeqScanPlan> GetSeqScanPlanA(bool bind_tve) {
+    auto *a_col_exp =
+        new expression::TupleValueExpression(type::TypeId::INTEGER, 0, 0);
+    if (bind_tve)
```
Comment (chenboy) [inline test/codegen/query_cache_test.cpp]: Please add brackets

File: test/codegen/query_cache_test.cpp [hunk 2]
```diff
@@ -51,6 +51,30 @@ class QueryCacheTest : public PelotonCodeGenTest {
...
+    auto *a_eq_40 = new expression::ComparisonExpression(
+        ExpressionType::COMPARE_EQUAL, a_col_exp, const_40_exp);
+    return std::shared_ptr<planner::SeqScanPlan>(new planner::SeqScanPlan(
+        &GetTestTable(TestTableId()), a_eq_40, {0, 1}));
+  }
+
+  std::shared_ptr<planner::SeqScanPlan> GetSeqScanPlanB(bool bind_tve) {
+    auto *b_col_exp =
+        new expression::TupleValueExpression(type::TypeId::INTEGER, 0, 1);
+    if (bind_tve)
```
Comment (chenboy) [inline test/codegen/query_cache_test.cpp]: Ditto, brackets.

File: test/codegen/query_cache_test.cpp [hunk 3]
```diff
@@ -51,6 +51,30 @@ class QueryCacheTest : public PelotonCodeGenTest {
         &GetTestTable(TestTableId()), a_gt_40, {0, 1}));
   }
 
+  std::shared_ptr<planner::SeqScanPlan> GetSeqScanPlanA(bool bind_tve) {
```
Comment (chenboy) [inline test/codegen/query_cache_test.cpp]: Please add some brief comments on what these helper functions generate

File: test/codegen/query_cache_test.cpp [hunk 4]
```diff
@@ -253,6 +278,71 @@ TEST_F(QueryCacheTest, SimpleCache) {
...
+  // clear the cache
+  codegen::QueryCache::Instance().Clear();
+  EXPECT_EQ(0, codegen::QueryCache::Instance().GetCount());
+
+  // execute SELECT b FROM table where b == 41;
+  codegen::BufferingConsumer buffer_2{{0, 1}, context_b};
+  CompileAndExecuteCache(plan_b, buffer_2, cached);
+
+  const auto &results_2 = buffer_2.GetOutputTuples();
+  EXPECT_EQ(1, results_2.size());
```
Comment (chenboy) [inline test/codegen/query_cache_test.cpp]: Do we also want to check the output tuple here?

File: test/codegen/query_cache_test.cpp [hunk 5]
```diff
@@ -253,6 +278,71 @@ TEST_F(QueryCacheTest, SimpleCache) {
...
+  EXPECT_EQ(1, codegen::QueryCache::Instance().GetCount());
+
+  // re-execute SELECT a FROM table where a == 40;
+  codegen::BufferingConsumer buffer_3{{0, 1}, context_a};
+  CompileAndExecuteCache(plan_a, buffer_3, cached);
+
+  // cache should not hit
+  EXPECT_FALSE(cached);
+  const auto &results_3 = buffer_3.GetOutputTuples();
+  EXPECT_EQ(1, results_3.size());
```
Comment (chenboy) [inline test/codegen/query_cache_test.cpp]: Same, do we also want to check the output tuple?

**Comment by nwang57** [comment]
@chenboy Hi, I have solved all the issues. Please have a look at it again, thanks.

**Comment by saatviks** [comment]
@chenboy could you take a look at this?

**Comment by chenboy** [comment]
I believe there're still unresolved comments. I think we may want to also check the content of the tuples returned.

**Comment by apavlo** [comment]
I will merge this when the build passes.

**Pull request merged**